ledger/README.md

```diff
@@ -7,3 +7,58 @@ This package provides two ledger implementations:
 - **Complete Ledger** implements a fast, memory-efficient and reliable ledger. It holds a limited number of recently used states in memory (for speed) and uses write-ahead logs and checkpointing to provide reliability. Under the hood complete ledger uses a collection of MTries(forest). MTrie is a customized in-memory binary Patricia Merkle trie storing payloads at specific storage paths. The payload includes both key-value pair and storage paths are determined by the PathFinder. Forest utilizes unchanged sub-trie sharing between tries to save memory.
 
 - **Partial Ledger** implements the ledger functionality for a limited subset of keys. Partial ledgers are designed to be constructed and verified by a collection of proofs from a complete ledger. The partial ledger uses a partial binary Merkle trie which holds intermediate hash value for the pruned branched and prevents updates to keys that were not part of proofs.
+
+## Definitions
+
+### binary Merkle tree
+
+![binary Merkle tree image](/ledger/docs/binary_merkle_tree.png?raw=true "binary Merkle tree" )
+
+In this context a *binary Merkle tree* is defined as a full binary tree with an specific height including three type of nodes:
```

Comment:
I think we should use the established term 
```suggestion
In this context a *binary Merkle tree* is defined as a [perfect binary tree](https://xlinux.nist.gov/dads/HTML/perfectBinaryTree.html) with a specific height including three type of nodes:
```

Reply:
updated at 0d1658b